packages/flutter_tools/lib/src/project.dart

```diff
@@ -682,7 +682,10 @@ class IosProject extends FlutterProjectPlatform implements XcodeBasedProject {
     .childDirectory('Flutter')
     .childFile('Generated.xcconfig');
 
-  Directory get compiledDartFramework => _flutterLibRoot
+  /// No longer compiled to this location.
+  ///
+  /// Used only for "flutter clean" to remove old references.
+  Directory get deprecatedCompiledDartFramework => _flutterLibRoot
```

Comment:
Open to naming suggestions here.

Reply:
Seems reasonable to me. Might want to TODO removal after a stable release cycle or so